diesel/src/query_source/aliasing/macros.rs

```diff
@@ -0,0 +1,128 @@
+/// Declare a new alias for a table
+///
+/// Example usage
+/// -------------
+/// ```rust
+/// # include!("../../doctest_setup.rs");
+/// fn main() {
+///     use schema::users;
+///     let connection = &mut establish_connection();
+///     let (users1, users2) = diesel::alias!(schema::users as user1, schema::users as user2);
+///     let res = users1
+///         .inner_join(users2.on(users1.field(users::id).eq(users2.field(users::id))))
+///         .select((users1.fields((users::id, users::name)), users2.field(users::name)))
+///         .order_by(users2.field(users::id))
+///         .load::<((i32, String), String)>(connection);
+///     assert_eq!(
+///         res,
+///         Ok(vec![
+///             ((1, "Sean".to_owned()), "Sean".to_owned()),
+///             ((2, "Tess".to_owned()), "Tess".to_owned()),
+///         ]),
+///     );
+/// }
+/// ```
+///
+///
+/// Make type expressable
+/// ---------------------
+/// It may sometimes be useful to declare an alias at the module level, in such a way that the type
+/// of a query using it can be expressed (to not declare it anonymously).
+///
+/// This can be achieved in the following way
+/// ```rust
+/// # include!("../../doctest_setup.rs");
+/// use diesel::{query_source::Alias, dsl};
+///
+/// diesel::alias!(schema::users as users_alias: UsersAlias);
```

Comment:
I would prefer something like:

```rust
const USER_ALIAS: Alias<user_alias> = diesel::alias!(schema::users as user_alias);
``` 

here if possible. Making the declared identifier visible, instead of hiding it behind a macro. That makes it easier to search for the corresponding definition as it looks like any other rust code.

Reply:
I don't think that would be possible because that would make this macro an expression macro so the `user_alias` type would be scoped to inside the construction of that expression at best.

My goal was to make it feel like `table!` on use (I had to use a const for that instead of a unit struct because of the alias wrapper).

Reply:
If we could not make it an expression maco, I would like to suggest the same solution as for `sql_function!` here. Just use `diesel::alias!(const MY_ALIAS_IDENT: UserAlias = schema::users as UserAlias);` (or something slightly different as long as it matches the normal rust const definition) to define an alias at  module level.

Reply:
So the writing would then probably be:
```rust
//                               const name             AliasSource                   alias name within the SQL query
diesel::alias!(const USERS_ALIAS: Alias<UsersAlias> = schema::users as users_alias);
```
I'm not sure the fact it's a `const` instead of e.g. a unit struct that can be instantiated implicitly is typically relevant to the end user - or what they would focus on.
I think in our codebase we'd favor the `table!`-like writing to the `fn`-like writing when declaring these at the module level.
@Elrendio wdyt? (vs ```rustdiesel::alias!(schema::users as users_alias: UsersAlias);`)

Reply:
Although maybe it would be relevant considering if you're declaring at the module level that might be because you need to express the types so making the magic not magic would help with that.

Reply:
For now I'm supporting both

Reply:
Hi :wave:

After discussing it IRL with @Ten0 I'm in favor of `diesel::alias!(schema::users as users_alias: UsersAlias);` and even better `diesel::alias!(schema::users as users_alias);` when you don't need to reference the type.

# Arguments:
1. I find `diesel::alias!(schema::users as users_alias);` simplier than `diesel::alias!(const USERS_ALIAS: Alias<UsersAlias> = schema::users as users_alias);` as there's less boiler plate. Seems to me it will be simplier for developers to adopt the feature
2. I don't think developpers need to know the internals of the feature before using it. Most people @Stockly for example will just want to be able to reference an alias in a query. 
4. Having the same writing for when using an alias in a function and in a module feels simplier to me. Notably, it will enable to just move the line of code out of the function.
5. We can support both writings for people that want to deep dive a bit more 

Have a nice day,
Elrendio

Reply:
TODO if we keep the `const` writing we probably want to use `static_cond!` to prevent from writing:
```rust
diesel::alias! {
    const USERS_ALIAS: Alias<UsersAlias> = schema::users as users_alias;
    const USERS_ALIAS_2: Alias<UsersAlias2> = schema::users as users_alias;
}
```